cfy_manager/components/rabbitmq/rabbitmq.py

```diff
@@ -40,172 +38,163 @@
-    _validate_rabbitmq_running()
-
-
-def install():
-    logger.notice('Installing RabbitMQ...')
-    _install()
-    _configure()
-    logger.notice('RabbitMQ successfully installed')
-
-
-def configure():
-    logger.notice('Configuring RabbitMQ...')
-    _configure()
-    logger.notice('RabbitMQ successfully configured')
-
-
-def remove():
-    logger.notice('Removing RabbitMQ...')
-    yum_remove('erlang')
-    logger.info('Stopping the Erlang Port Mapper Daemon...')
-    sudo(['epmd', '-kill'], ignore_failures=True)
-    systemd.remove(RABBITMQ, service_file=False)
-    yum_remove('socat')
-    logger.notice('RabbitMQ successfully removed')
-
-
-def start():
-    logger.notice('Starting RabbitMQ...')
-    systemd.start(RABBITMQ)
-    _validate_rabbitmq_running()
-    logger.notice('RabbitMQ successfully started')
-
-
-def stop():
-    logger.notice('Stopping RabbitMQ...')
-    systemd.stop(RABBITMQ)
-    logger.notice('RabbitMQ successfully stopped')
+    def _start_rabbitmq(self):
+        logger.info("Starting RabbitMQ Service...")
+        # rabbitmq restart exits with 143 status code that is valid in this case.
+        systemd.restart(RABBITMQ, ignore_failure=True)
+        wait_for_port(SECURE_PORT)
+        self._set_policies()
+        systemd.restart(RABBITMQ)
+
+    def _validate_rabbitmq_running(self):
+        logger.info('Making sure RabbitMQ is live...')
+        systemd.verify_alive(RABBITMQ)
+
+        result = self._rabbitmqctl(['status'])
+        if result.returncode != 0:
+            raise ValidationError('Rabbitmq failed to start')
+
+        if not is_port_open(SECURE_PORT, host='127.0.0.1'):
+            raise NetworkError(
+                '{0} error: port {1}:{2} was not open'.format(
+                    RABBITMQ, '127.0.0.1', SECURE_PORT)
+            )
+
+    def _configure(self):
+        systemd.configure(RABBITMQ)
+        self._init_service()
+        self._delete_guest_user()
+        self._create_rabbitmq_user()
+        self._start_rabbitmq()
+        self._validate_rabbitmq_running()
+
+    def install(self):
+        logger.notice('Installing RabbitMQ...')
+        self._install()
+        self._configure()
+        logger.notice('RabbitMQ successfully installed')
+
+    def configure(self):
+        logger.notice('Configuring RabbitMQ...')
+        self._configure()
+        logger.notice('RabbitMQ successfully configured')
+
+    def remove(self):
+        logger.notice('Removing RabbitMQ...')
+        yum_remove('erlang')
+        logger.info('Stopping the Erlang Port Mapper Daemon...')
+        sudo(['epmd', '-kill'], ignore_failures=True)
+        systemd.remove(RABBITMQ, service_file=False)
+        yum_remove('socat')
+        logger.notice('RabbitMQ successfully removed')
+
+    def start(self):
+        logger.notice('Starting RabbitMQ...')
+        systemd.start(RABBITMQ)
+        self._validate_rabbitmq_running()
+        logger.notice('RabbitMQ successfully started')
+
+    def stop(self):
```

Comment:
Can be moved to the base component

Reply:
Will do after merge as it's not related to this feature specifically